File: 2003/CS101SamplesAll/CS101SamplesWebDevelopment/DataAccess/App_Code/ProductCollection.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Collections;

namespace Microsoft.Samples.Web.DataAccess
{
	/// <summary>
	/// Summary description for ProductCollection
	/// </summary>
	public class ProductCollection : CollectionBase
	{
		public ProductCollection()
		{
		}

		public Product Item(int index)
		{
			return (Product)List[index];
		}

		public int Add(Product product)
		{
			return List.Add(product);
		}
	}
}

File: 2003/VB101SamplesAll/VB101SamplesWebDevelopment/DataAccess/App_Code/ProductCollection.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Collections
Namespace Microsoft.Samples.Web.DataAccess

    Public Class ProductCollection
        Inherits CollectionBase

        Public Sub New()
        End Sub

        Public Function Item(ByVal index As Integer) As Product
            Return CType(List(index), Product)
        End Function

        Public Function Add(ByVal product As Product) As Integer
            Return List.Add(product)
        End Function
    End Class
End Namespace
